### DISCUSSION TITLE
Question: is it possible to make CPR apply an affliction on success?

### DISCUSSION THREAD

**Discussion by Krzeszny** [FakeFishGames/Barotrauma#7764] [Code questions and modding support]
The config for CPR has this:
```
  <CPRSettings
    revivechanceperskill=""
    revivechanceexponent=""
    revivechancemin=""
    revivechancemax=""
    stabilizationperskill=""
    stabilizationmin=""
    stabilizationmax=""
    damageskillthreshold=""
    damageskillmultiplier=""
    insufficientskillaffliction="">
  </CPRSettings>
```
It's possible to apply an affliction when CPR **fails**, What about the other outcome?

**Comment by LePiromano** [comment]
don't we specifically have a section in discussions for these questions

**Reply by ekqvist** [reply]
I've converted this into a Discussion now

**Reply by Krzeszny** [reply]
There _is_ a Question tag in Issues, that's why I posted it there. Now that you've converted it into a discussion, I can practically guarantee nobody will answer.

**Reply by Ek-Videogames** [reply]
### #Won'tFix Moment
Short answer, no and/or hardcoded. 

Revive chance is the hardcoded effect that instantly heals when it triggers.
The 'stabilization chance min/max' does not really give oxygen as one would expect given the comment saying:
`"Stabilization means that the oxygen level of the target increases"`

There is no 'on success' status effect type for CPR, and 'on failure' won't let you set a custom status effect only apply an affliction.
( By setting a 'dummy' affliction which then acts on its own status effects is the workaround used for the CPR mod. )
